src/server/pps/server/poller.go

```diff
@@ -176,25 +172,27 @@ func (m *ppsMaster) pollPipelines(ctx context.Context) {
 // to CRASHING
 func (m *ppsMaster) pollPipelinePods(ctx context.Context) {
 	if err := backoff.RetryUntilCancel(ctx, backoff.MustLoop(func() error {
-		kubePipelineWatch, err := m.a.env.KubeClient.CoreV1().Pods(m.a.namespace).Watch(
-			ctx,
-			metav1.ListOptions{
-				LabelSelector: metav1.FormatLabelSelector(metav1.SetAsLabelSelector(
-					map[string]string{
-						"component": "worker",
-					})),
-				Watch: true,
-			})
+		watch, cancel, err := m.kd.WatchPipelinePods(ctx)
 		if err != nil {
 			return errors.Wrap(err, "failed to watch kubernetes pods")
 		}
-		defer kubePipelineWatch.Stop()
+		defer cancel()
 	WatchLoop:
 		for {
 			select {
 			case <-ctx.Done():
 				return nil
-			case event := <-kubePipelineWatch.ResultChan():
+			case event, ok := <-watch:
+				if !ok {
+					log.Warn("kubernetes pod watch unexpectdly ended - restarting watch")
+					cancel()
+					watch, cancel, err = m.kd.WatchPipelinePods(ctx)
+					if err != nil {
+						return errors.Wrap(err, "failed to watch kubernetes pods")
+					}
+					defer cancel()
+					continue
```

Comment:
I think I prefer just logging and returning `ErrContinue` here unless there's an obstacle I'm not seeing